### DISCUSSION TITLE
Go To Definition Not Working in Decompiled Class Files

### DISCUSSION THREAD

**Discussion by Corbanator** [mfussenegger/nvim-jdtls#646] [Q&A]
I have a function `on_attach` that maps LSP actions, including the `vim.lsp.buf.defintion` action. I refer to this function in the following block:

```
mason_lspconfig.setup_handlers {
  function(server_name)
    require('lspconfig')[server_name].setup {
      capabilities = capabilities,
      on_attach = on_attach,
      settings = servers[server_name],
    }
  end,
}
```

This works flawlessly while in java files, but when I use it to jump into a decompiled class file from a library, these keymaps no longer show up. The keybind I normally have set to `vim.lsp.buf.defintion`, `gd`, seems to instead do its best to jump within the file, even though it doesn't seem that it should be mapped. Usually this leads to an import statement. Manually calling `:lua vim.lsp.buf.definition()` works flawlessly, which indicates to me that the keybind is the issue. Keybinds I specify for only Java seem to also still be there.

Is there something I'm doing wrong in my configuration, or an easy way to get these keybinds to work in decompiled class files without manually adding them to my `ftplugin/java.lua`?

**Comment by CarlosLlaque** [comment]
Hi, have you found a solution?
